internal/util/password/scramsha256_test.go

```diff
@@ -0,0 +1,230 @@
+			"serverKey", "cLmYEp4e6nRZDv4vrrpjYSt/FPP/Ekt/XVZVoDlrByw=",
+		)),
+	},
+	{
+		params: scramSHA256Params{
+			iterationCount: 2,
+			saltLen:        4,
+		},
+		username: "username",
+		password: "password",
+		salt:     []byte("salt"),
+		want: must.NotFail(types.NewDocument(
+			"storedKey", "db2Cdby2HHY1enQpujvJPfRJNlLyQ95MIEMwybJdFcI=",
+			"iterationCount", int32(2),
+			"salt", "c2FsdA==",
+			"serverKey", "a0OWicFaTNUVr7ZJDEnGc0sn9GLSAUyannq6uYeSJRs=",
+		)),
+	},
+	{
+		params: scramSHA256Params{
+			iterationCount: 4096,
+			saltLen:        4,
+		},
+		username: "username",
+		password: "password",
+		salt:     []byte("salt"),
+		want: must.NotFail(types.NewDocument(
+			"storedKey", "lF4cRm/Jky763CN4HtxdHnjV4Q8AWTNlKvGmEFFU8IQ=",
+			"iterationCount", int32(4096),
+			"salt", "c2FsdA==",
+			"serverKey", "ub8OgRsftnk2ccDMOt7ffHXNcikRkQkq1lh4xaAqrSw=",
+		)),
+	},
+	{
+		params: scramSHA256Params{
+			iterationCount: 4096,
+			saltLen:        36,
+		},
+		username: "username",
+		password: "passwordPASSWORDpassword",
+		salt:     []byte("saltSALTsaltSALTsaltSALTsaltSALTsalt"),
+		want: must.NotFail(types.NewDocument(
+			"storedKey", "kl1yVUP4s3BJMFrtaC4zLJycbv6k5yMBhVgocYmsYsU=",
+			"iterationCount", int32(4096),
+			"salt", "c2FsdFNBTFRzYWx0U0FMVHNhbHRTQUxUc2FsdFNBTFRzYWx0",
+			"serverKey", "4QTMss7Dzi+pk8C8cyql++OaWqI0y/FyhXJI7W9acHI=",
+		)),
+	},
+	{
+		params: scramSHA256Params{
+			iterationCount: 4096,
+			saltLen:        5,
+		},
+		username: "username",
+		password: "pass\x00word",
+		salt:     []byte("sa\x00lt"),
+		err:      "prohibited character",
+	},
+	{
+		params: scramSHA256Params{
+			iterationCount: 1,
+			saltLen:        4,
+		},
+		username: "username",
+		password: "passwd",
+		salt:     []byte("salt"),
+		want: must.NotFail(types.NewDocument(
+			"storedKey", "dcwmgrDYICpRpKjwHLKxZ/21/go62U106s5V4i9v+Q8=",
+			"iterationCount", int32(1),
+			"salt", "c2FsdA==",
+			"serverKey", "qFes4m5Z84MaC2hSJqCR2e/FBz7goMVu/RTRNnb5Fj0=",
+		)),
+	},
+	{
+		params: scramSHA256Params{
+			iterationCount: 80000,
+			saltLen:        4,
+		},
+		username: "username",
+		password: "Password",
+		salt:     []byte("NaCl"),
+		want: must.NotFail(types.NewDocument(
+			"storedKey", "EI8wmB+eWKGZ8k+dln75YQnX8lj+MBoG6+eH9AzR1e4=",
+			"iterationCount", int32(80000),
+			"salt", "TmFDbA==",
+			"serverKey", "Wbwo6JsaJrZ/1Bf7F+45jY2VURuezLXxADxUuzWdZ/4=",
+		)),
+	},
+	{
+		params: scramSHA256Params{
+			iterationCount: 4096,
+			saltLen:        5,
+		},
+		username: "username",
+		password: "Password",
+		salt:     []byte("sa\x00lt"),
+		want: must.NotFail(types.NewDocument(
+			"storedKey", "BHJbwIZ9YCvb+dFApByBLe4jR7gquvBm3kPApnWCylk=",
+			"iterationCount", int32(4096),
+			"salt", "c2EAbHQ=",
+			"serverKey", "qpmMln7z50yoT66R1lT55pUvBxu8BqZU4X8dRp38NWo=",
+		)),
+	},
+}
+
+func TestSCRAMSHA256(t *testing.T) {
+	for i, tc := range scramSHA256TestCases {
+		i, tc := i, tc
+		t.Run(fmt.Sprintf("%d", i), func(t *testing.T) {
```

Comment:
Just a suggestion :hugs: To identify failing test case easily, we use often `map[string]struct` for test cases and name each test, that way we can identify failing test with name instead of an arbitrary number. Also allow us to see what each test case is testing from the name
```suggestion
		name, tc := name, tc
		t.Run(name, func(t *testing.T) {
```